model/workflow.go

```diff
@@ -486,12 +494,45 @@ func (e *End) UnmarshalJSON(data []byte) error {
 type ContinueAs struct {
 	WorkflowRef
 	// TODO: add object or string data type
-	// If string type, an expression which selects parts of the states data output to become the workflow data input of continued execution. If object type, a custom object to become the workflow data input of the continued execution
+	// If string type, an expression which selects parts of the states data output to become the workflow data input of
+	// continued execution. If object type, a custom object to become the workflow data input of the continued execution
 	Data interface{} `json:"data,omitempty"`
 	// WorkflowExecTimeout Workflow execution timeout to be used by the workflow continuing execution. Overwrites any specific settings set by that workflow
 	WorkflowExecTimeout WorkflowExecTimeout `json:"workflowExecTimeout,omitempty"`
 }
 
+func (c *ContinueAs) UnmarshalJSON(data []byte) error {
```

Comment:
Should we change this style unmarshal to https://github.com/serverlessworkflow/sdk-go/blob/main/model/state_exec_timeout.go#L35 style？

Reply:
tried something similar, not sure why it doesn't set the data and the timeout values.